lib/dbservice_web/controllers/group_user_controller.ex

```diff
@@ -121,4 +122,36 @@ defmodule DbserviceWeb.GroupUserController do
       send_resp(conn, :no_content, "")
     end
   end
+
+  defp create_new_group_user(conn, params) do
+    group = Groups.get_group!(params["group_id"])
+
+    enrollment_record = %{
+      "group_id" => group.child_id,
+      "group_type" => group.type,
+      "user_id" => params["user_id"],
+      "grade_id" => params["grade_id"],
+      "academic_year" => params["academic_year"],
+      "start_date" => params["start_date"]
+    }
```

Comment:
When creating a `group_user`, I don't feel we should also be sending `enrollment_record` fields as well like `academic_year`, `grade_id` etc.

Reply:
Why not?

Reply:
It violates Single Responsibility Principle

Reply:
I feel like each endpoint should have a clear, well-defined responsibility, adhering to the SRP.

Reply:
Yeah, the group-user endpoint creates both a group-user record and an ER. 
If we remove this creation, then its the responsibility of the external agency to create both records separately.

Reply:
As per discussion on discord, letting this be.